protocol/Cargo.toml

```diff
@@ -10,15 +10,18 @@ rust-version = "1.63.0"
 
 [features]
 default = ["std"]
+# High-level wrappers using futures traits.
 async = ["std", "futures/std"]
```

Comment:
Sounds like this feature should've been called `futures` and use the `dep:` syntax.

Reply:
Is `futurues` a more conventional name for the async interface of a library? That is what this feature flag exposes, which I think is a little more than just futures.

Reply:
No, the convention is that if you clearly use a *public* dependency then the feature should be named after the dependency. E.g. we have `serde` feature in other crates that activates the `serde` dependency and whatever else is required to implement it. This is even in the API guidelines. If the dependency is private and you're not sure if it'll even stay then using a different name is fine. (Here, it's public but sadly, `futures` is unstable.)

Reply:
I found these [API guidelines on feature names](https://rust-lang.github.io/api-guidelines/naming.html#feature-names-are-free-of-placeholder-words-c-feature), but they don't mention exactly what to base the name on. Although, lookin around the ecosystem, I do see the convention that they generally are tied to a dependency's name. I don't feel too terrible about this name though since the intention was to capture "async capabilities without specifying a particular runtime", and to me "futures" would be kinda confusing since the intention isn't to directly interact with futures...but will definitely keep this in mind for naming things next time.

Reply:
I remember being in some discussion around this and the desire to name the features the same even motivated some changes to `cargo`. So it's definitely a thing even if we can't find a specific doc saying so.

I haven't looked into the code, but if you don't use `futures` publicly (their types in the API) then `async` should be fine and maybe even really a better name. But if it's public I'd recommend renaming it.

Reply:
The types are used as trait bounds on the input parameters for public functions, so I think that counts? I can rename it in that case in another patch.

Reply:
Yeah, rename would be appropriate in that case.